internal/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -300,7 +307,18 @@ func (r *Reconciler) reconcile(ctx context.Context, logger logr.Logger, cluster
 		}
 		errList := []error{}
 		for _, t := range append(healthy, unhealthy...) {
```

Comment:
Not new with this PR, but I'm really confused why we are patching all Machines twice (if I'm reading this correctly)

Reply:
My current understanding is that we are patching here only if remediation is not allowed (and we patch both unhealthy and healthy machines).

The other two patch machines are for when remediation is allowed, one for healthy machines, the other for unhealthy.

But if you want we can take a deeper look together

Reply:
Let's take a look together later. To me it looks like some of these patches are always no-ops